### PR TITLE
feat: get only user apps in file-movement for performance aspect, add comments

### PR DESCRIPTION
**Pull request by KazuCocoa**
fixes https://github.com/appium/appium/issues/12981

### CODE REVIEW

**Comment by KazuCocoa** [comment]
cc @umutuzgur 
I just created a PR.

I checked file-management, too.
The target apps should have `UIFileSharingEnabled`. So far, I could not find the attribute in system app.
So, we probably can ignore "Sytem" for file-management method.
(I just noted it)

**Comment by KazuCocoa** [comment]
https://github.com/appium/appium-ios-device/pull/38/files is pretty good.
Will leave only comment in this PR.

File: lib/commands/file-movement.js [hunk 1]
```diff
@@ -318,13 +318,16 @@ async function pullFromRealDevice (device, remotePath, isFile) {
...
- * @returns {Array<string>} A list of bundle ids
+ * @returns {Array<string>} A list of User level apps' bundle ids which has
+ *                          'UIFileSharingEnabled' attribute.
+ *                          Only user apps might have it.
  */
 async function getAvailableBundleIds (udid) {
   const service = await services.startInstallationProxyService(udid);
   try {
-    const applications = await service.listApplications();
+    const applications = await service.listApplications({applicationType: 'User'});
```
Comment (KazuCocoa) [inline lib/commands/file-movement.js]: I haven't find UIFileSharingEnabled in system apps so far.
We can ignore it, for now, to improve find app performance.

Comment (umutuzgur) [reply lib/commands/file-movement.js]: Good idea

File: lib/ios-deploy.js [hunk 2]
```diff
@@ -100,6 +100,18 @@ class IOSDeploy {
...
+   * The target bundleid can be User and System apps.
+   * @param {string} bundleid The bundleId to ensure it is installed
+   * @return {boolean} Returns True if the bundleid exists in the result of 'listApplications' like:
+   * { "com.apple.Preferences":{
+   *   "UIRequiredDeviceCapabilities":["arm64"],
+   *   "UIRequiresFullScreen":true,
+   *   "CFBundleInfoDictionaryVersion":"6.0",
+   *   "Entitlements":
+   *     {"com.apple.frontboard.delete-application-snapshots":true,..
+   */
```
Comment (umutuzgur) [inline lib/ios-deploy.js]: I can actually speed this part up by using a different API. We can directly ask the phone if a certain bundle id is present. I will try to implement that today

**Pull request merged**